### PR TITLE
frontend/#1233/hotspot, referral native changes and added geocode co…

### PR DESCRIPTION
**Pull request by ashishsingh101**
…nfig

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

1. added geocoder strictBounds, 50km for Bangalore and 100km for other cities
2. added native changes for hotspot and referral deep link support

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 1]
```diff
@@ -630,6 +630,9 @@ protected Vector<String> handleDeepLinkIfAvailable(Intent appLinkIntent){
                 if (key.equals("vp")){
                     viewParam = query_params.get(key);
                     break;
+                } else if (key.equals("referrer")) {
```
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:633]: we should use smaller query params to keep the length of links shorter. Like rf here for referrer/referral

File: Frontend/ui-customer/src/Helpers/Utils.purs [hunk 2]
```diff
@@ -729,4 +730,18 @@ getExistingTags savedLoc = map (\item -> DS.toLower $ item.tag) savedLoc
 -- , recencyDate : Nothing
 -- , locationScore : Nothing
 
--- }) savedLocation )
\ No newline at end of file
+-- }) savedLocation )
+
+getCityConfig :: Array CityConfig -> String -> CityConfig
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Helpers/Utils.purs:735]: these config functions should be there in config related files instead of utils?

**Review by witcher-shailesh — COMMENTED** [review]
rename geocoder with geocode across the changes

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java (hunk 1 above)
Comment (ashishsingh101) [reply Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:633]: yes, but in this case we are using similar link for playStore referral and deepLink where playStore accepts query as `referrer`

**Pull request merged**